### PR TITLE
For #17917: Use View binding in Share views

### PR DESCRIPTION
**Pull request by codrut-topliceanu**
For https://github.com/mozilla-mobile/fenix/issues/17917

### Pull Request checklist

- [x] **Tests**: This PR includes no tests as it is only a minor change
- [x] **Screenshots**: This PR does not include screenshots as no user visible changes have been made.
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md).

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/share/ShareCloseView.kt [hunk 1]
```diff
@@ -28,13 +27,16 @@ class ShareCloseView(
     val adapter = ShareTabsAdapter()
 
     init {
-        LayoutInflater.from(containerView.context)
-            .inflate(R.layout.share_close, containerView, true)
+        val binding = ShareCloseBinding.inflate(
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/share/ShareCloseView.kt:30]: nit: Let's remove the `LayoutContainer` :)

File: app/src/test/java/org/mozilla/fenix/share/ShareCloseViewTest.kt [hunk 2]
```diff
@@ -26,18 +26,19 @@ class ShareCloseViewTest {
 
     @Before
     fun setup() {
-        container = FrameLayout(testContext)
+        container = ConstraintLayout(testContext)
```
Comment (Mugurell) [inline app/src/test/java/org/mozilla/fenix/share/ShareCloseViewTest.kt:29]: nit: Is this change needed?

**Review by Mugurell — APPROVED** [review]

File: app/src/test/java/org/mozilla/fenix/share/ShareCloseViewTest.kt (hunk 2 above)
Comment (codrut-topliceanu) [reply app/src/test/java/org/mozilla/fenix/share/ShareCloseViewTest.kt:29]: Yes.

image
